lib/AST/NameLookup.cpp

```diff
@@ -1450,6 +1450,16 @@ static bool isAcceptableLookupResult(const DeclContext *dc,
     }
   }
 
+  // We don't support nested types in generics yet.
+  if (auto *NTD = dyn_cast<NominalTypeDecl>(decl)) {
```

Comment:
@DougGregor @CodaFi I need some help with this since I don't really know what is the best place to filter these out, it seems like such members shouldn't be returned from `getMembers()` so they shouldn't loaded into a lookup table, but I couldn't figure out how member loading works exactly because none of the request break points I set were triggered.